crates/paralegal-policy/src/context.rs

```diff
@@ -322,7 +322,18 @@ impl Context {
 
     /// Returns whether the given Node has the marker applied to it directly or via its type.
     pub fn has_marker(&self, marker: Marker, node: Node) -> bool {
-        // TODO: check the type as well
+        if let Some(src) = node.typ.as_data_source() {
+            if let Some(typ) = &self.desc.controllers[&node.ctrl_id].types.get(&src) {
```

Comment:
We should probably add a convenience function for getting the type(s) of a node.

Reply:
What do you mean? So as_data_source etc functions on the node itself that just called it on the typ field?

Reply:
No I mean the essentially this expression `self.desc.controllers[&node.ctrl_id].types.get(&src)`, the Rust type if you will.

Reply:
done